Kickstarter-iOS/Views/SettingsTableViewHeader.swift

```diff
@@ -0,0 +1,21 @@
+import Library
+import Prelude
+
+final class SettingsTableViewHeader: UITableViewHeaderFooterView {
```

Comment:
Looks like there's still some confusion between when to use `UIView` and when to use `UITableViewHeaderFooterView` - https://github.com/kickstarter/ios-oss/pull/570#discussion_r254473087.

As per the documentation the `UITableViewHeaderFooterView` should only be used as a reusable section header / footer - https://developer.apple.com/documentation/uikit/uitableviewheaderfooterview. 

On the other hand `UITableView`'s header / footer is simply an accessory view (non-reusable).

The difference is the reusability (which we don't really need in case of `tableHeaderView`/`tableFooterView` there's always going to be just one instance. The section headers/footers are different though.

In order to use `UIView` subclass as a `tableHeaderView` it looks like we need to do the following..

1. Convert `SettingsTableViewHeader` to a `UIView` subclass
2. Remove `|> \.preferredMaxLayoutWidth .~ self.titleLabel.bounds.width` from `SettingsTableViewHeader` (auto layout will do this for us)
3. Simply assign the header view

```
// We no longer need this since this is mostly for section header/footer views
self.tableView.registerHeaderFooter(nib: .SettingsTableViewHeader)
```

in `viewDidLoad` set the `tableView.tableHeaderView` and add width constraint so that it's properly resizing based on the screen size

```
if let tableHeaderView = SettingsTableViewHeader.fromNib(nib: Nib.SettingsTableViewHeader) {
  tableHeaderView.translatesAutoresizingMaskIntoConstraints = false
  tableHeaderView.configure(with: Strings.Any_payment_methods_you_saved_to_Kickstarter())

  self.tableView.tableHeaderView = tableHeaderView

  NSLayoutConstraint.activate([
    tableHeaderView.widthAnchor.constraint(equalTo: self.tableView.widthAnchor)
    ])
  }
```

Remove `viewDidLayoutSubviews` since it's no longer necessary. I've tried to follow these articles but didn't have much luck of properly resizing without setting the `width` constraint.
https://useyourloaf.com/blog/variable-height-table-view-header/
https://collindonnell.com/2015/09/29/dynamically-sized-table-view-header-or-footer-using-auto-layout/

🤷‍♂️ 

The above should remove a lot of boiler plate and we're also following best practices of how to use `tableHeaderView`/`tableFooterView` so it should be future proof should Apple made any changes to the way headers/footers are presented.

Same applies to the footer view.

Reply:
I agree it's not necessary to make these `UITableViewHeaderFooterView`s 👍  However I was never able to get the accessory views to properly size using only autolayout. Maybe we can pair on it on Monday.

Reply:
Sounds good...the trick is having the `headerView` size itself (by setting its subview's auto layout - in our case just the label)..and then establishing the width constraint as outlined above.

Lets pair on that when you have a minute.